### PR TITLE
Update graph.md

### PR DESCRIPTION
**Pull request by Sigfried**
Starting by just gathering ideas in https://github.com/jhu-bids/TermHub/blob/subgraph-refactor-again/docs/graph.md

### CODE REVIEW

File: docs/graph.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
 # Documentation about vocabulary hierarchy algorithms and display
+## 2024-01-03 update
+The code for filling in gaps in concept subgraphs and displaying them as trees has never worked quite right -- though it usually is or seems right enough for people not to notice. But there have been a wide variety of edge cases like
+- The gaps aren't filled in correctly
+- Unneeded ancestors are added
+- The subgraph gap-filling or backend tree construction takes so long that it hangs
+- There's so many indented tree nodes that the browser slows to a crawl or crashes
+
+I'm realizing again that there are problems in the algorithm, so I'm going to use this space to try to plan out and document a new algorithm.
```
Comment (joeflack4) [inline docs/graph.md:9]: I guess I don't feel so bad now about my pre-networkx, pure Python based recursive tree building not working correctly. This is hard stuff!

File: docs/graph.md [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+
+### Steps
+1. Find all roots of subgraph (
+
+
+## Graph visualization
+I'd like to make a few test cases that include all the various edge cases that have to be handled. At least one small one that can be manually tinkered with and easy to understand and some big ones made with real data for testing front/back end performance and manageability and summarization features (collapsing many nodes into an expandable summary node.)
```
Comment (joeflack4) [inline docs/graph.md:16]: >  few test cases that include all the various edge cases that have to be handled

good idea

File: docs/graph.md [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+
+I would like to be able to use [graphviz and dot](https://graphviz.org/) for authoring and graphically rendering the small use cases, but it has a weird architecture. It has APIs, declarative language, CLIs, and stuff; but they rely on [executables](https://graphviz.org/download/) which are environment-specific. (See stack overflow for [how to handle azure](https://stackoverflow.com/questions/76244214/how-to-fix-pygraphviz-program-dot-not-found-in-path-error-in-azure-or-gcp).)
```
Comment (joeflack4) [inline docs/graph.md:18]: > graphviz and dot

I agree, these are nice-to-haves. But that's my worry, is that they are nice-to-haves and not needs. I just get the tingle in my head telling me this may be better to add later, what with all of the learning / architectural overhead.
